lib/network/modules/components/Edge.js

```diff
@@ -510,7 +534,12 @@ class Edge {
     this.drawLabel (ctx, viaNode);
   }
 
-
+  /**
+   *
+   * @param {CanvasRenderingContext2D} ctx
+   * @param {Object} arrowData
+   * @param {{toArrow: boolean, toArrowScale: (allOptions.edges.arrows.to.scaleFactor|{number}|allOptions.edges.arrows.middle.scaleFactor|allOptions.edges.arrows.from.scaleFactor|Array|number), toArrowType: *, middleArrow: boolean, middleArrowScale: (number|allOptions.edges.arrows.middle.scaleFactor|{number}|Array), middleArrowType: (allOptions.edges.arrows.middle.type|{string}|string|*), fromArrow: boolean, fromArrowScale: (allOptions.edges.arrows.to.scaleFactor|{number}|allOptions.edges.arrows.middle.scaleFactor|allOptions.edges.arrows.from.scaleFactor|Array|number), fromArrowType: *, arrowStrikethrough: (*|boolean|allOptions.edges.arrowStrikethrough|{boolean}), color: undefined, inheritsColor: (string|string|string|allOptions.edges.color.inherit|{string, boolean}|Array|*), opacity: *, hidden: *, length: *, shadow: *, shadowColor: *, shadowSize: *, shadowX: *, shadowY: *, dashes: (*|boolean|Array|allOptions.edges.dashes|{boolean, array}), width: *}} values
```

Comment:
Understood. I also find this an eyesore and potentially unmaintainable.

> ...and it probably makes more sense to make getFormattingValues return a proper class...

Ah, right, so this is all output from `getFormattingValues()`? Well, then your idea of making this a distinct class makes sense, at least for creating a proper doc reference.
`getFormattingValues()` is an internal method, not part of the API; so it doesn't need to be *that* well documented. Eternal users shouldn't be using it.